api/internal/handler/route_export/route_export.go

```diff
@@ -0,0 +1,357 @@
+		}
+		if route.ServiceID != nil {
+			extensions["x-apisix-serviceID"] = route.ServiceID
+		}
+		if route.UpstreamID != nil {
+			extensions["x-apisix-upstreamID"] = route.UpstreamID
+		}
+		if route.Script != nil {
+			extensions["x-apisix-script"] = route.Script
+		}
+		if route.ServiceProtocol != "" {
+			extensions["x-apisix-serviceProtocol"] = route.ServiceProtocol
+		}
+		if route.Vars != nil {
+			extensions["x-apisix-vars"] = route.Vars
+		}
+		// analysis route.URIs
+		for _, uri := range route.Uris {
+			if strings.Contains(uri, "*") {
+				paths[strings.Split(uri, "*")[0]+"{params}"] = pathItem
+				// add params introduce
+				paramsRefs = append(paramsRefs, &openapi3.ParameterRef{
+					Value: &openapi3.Parameter{
+						In:          "path",
+						Name:        "params",
+						Required:    true,
+						Description: "params in path",
+						Schema:      &openapi3.SchemaRef{Value: &openapi3.Schema{Type: "string"}}}})
+			} else {
+				paths[uri] = pathItem
+			}
+		}
+		if route.Plugins != nil {
+			param := &openapi3.Parameter{}
+			secReq := &openapi3.SecurityRequirements{}
+			plugins := make(map[string]interface{})
+			// analysis plugins
+			for key, value := range route.Plugins {
+				// analysis proxy-rewrite plugin
+				//if key == "proxy-rewrite" {
+				//	continue
+				//}
+
+				// analysis request-validation plugin
+				if key == "request-validation" {
+					for k, v := range value.(map[string]interface{}) {
+						if k == "header_schema" {
+							param.In = "header"
+							requestValidation := &entity.RequestValidation{}
+							reqBytes, _ := json.Marshal(&v)
+							err := json.Unmarshal(reqBytes, requestValidation)
+							if err != nil {
+								log.Errorf("json marshal failed: %s", err)
+							}
+							for key1, value1 := range requestValidation.Properties.(map[string]interface{}) {
+								for _, arr := range requestValidation.Required {
+									if arr == key1 {
+										param.Required = true
+									}
+								}
+								param.Name = key1
+								typeStr := value1.(map[string]interface{})
+								schema := &openapi3.Schema{Type: typeStr["type"].(string)}
+								param.Schema = &openapi3.SchemaRef{Value: schema}
+								paramsRefs = append(paramsRefs, &openapi3.ParameterRef{Value: param})
+							}
+						}
+						if k == "body_schema" {
+							m := map[string]*openapi3.MediaType{}
+							reqBytes, _ := json.Marshal(&v)
+							schema := &openapi3.Schema{}
+							err := json.Unmarshal(reqBytes, schema)
+							if err != nil {
+								log.Errorf("json marshal failed: %s", err)
+							}
+							for _, va := range route.Vars.([]interface{}) {
+								compile := regexp.MustCompile("^http_.*")
+								match := compile.Match([]byte(va.([]interface{})[0].(string)))
+								if match {
+									m[va.([]interface{})[2].(string)] = &openapi3.MediaType{Schema: &openapi3.SchemaRef{Value: schema}}
+									requestBody.Content = m
+								}
+							}
+						}
+					}
+					continue
+				}
```

Comment:
@Jaycean If only caring about the keys that equals to  `body_schema` and `header_schema`, IMHO, there is no need to iterate the entire map. Just use 

```go
if valueMap, ok := value.(map[string]interface{}); ok {
    if hsVal, ok := valueMap["header_schema"]; ok {
        // Deal with header schema
    }

    if bsVal, ok := valueMap["body_schema"]; ok {
        // Deal with body schema
    }
}

```

BTW: It would be better to check every type assertion in Golang.

Reply:
Thks, I'll check and change the code right away.